frontend/src/features/mapSideBar/SidebarContents/subforms/ClassificationForm.test.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import { ClassificationForm } from './ClassificationForm';
+import renderer from 'react-test-renderer';
+import { Formik } from 'formik';
+import { CLASSIFICATIONS } from 'constants/classifications';
+import { cleanup, fireEvent, render, wait } from '@testing-library/react';
+import { noop } from 'lodash';
+
+const mockClassifications = [
+  { value: CLASSIFICATIONS.CoreOperational, label: 'Core Operational' },
+  { value: CLASSIFICATIONS.CoreStrategic, label: 'Core Strategic' },
+  { value: CLASSIFICATIONS.SurplusEncumbered, label: 'Surplus Encumbered' },
+];
+
+it('renders correctly', () => {
+  const tree = renderer.create(form).toJSON();
+  expect(tree).toMatchSnapshot();
+});
+
+const form = (
+  <Formik onSubmit={noop} initialValues={{ classificationId: '' }}>
+    <ClassificationForm classifications={mockClassifications} field="classificationId" />
+  </Formik>
+);
+
+describe('field definition conditional render', () => {
+  it('renders definition when provided', () => {
+    const { getByText } = render(
+      <Formik onSubmit={noop} initialValues={{ classificationId: '' }}>
+        <ClassificationForm
+          classifications={mockClassifications}
+          field="classificationId"
+          fieldDescription="test description"
+        />
+      </Formik>,
+    );
+    expect(getByText('test description')).toBeInTheDocument();
+  });
+});
+
+describe('renders definitions correctly', () => {
+  afterEach(() => {
+    cleanup();
+    jest.clearAllMocks();
+  });
+  it('renders default defintion', () => {
+    const { getByText } = render(form);
+    expect(getByText(/select a classification/i)).toBeInTheDocument();
+  });
+  it('renders core operational definition', async () => {
+    const { getByText, container } = render(form);
+    const classificationId = container.querySelector('select[name="classificationId"]');
+    await wait(() => {
+      fireEvent.change(classificationId!, {
+        target: {
+          value: CLASSIFICATIONS.CoreOperational,
+        },
+      });
+    });
+    expect(getByText(/core Operational – assets that are functionally/i)).toBeInTheDocument();
```

Comment:
does the first character have to lower case for this to work or is it case insensitive?

Reply:
Normally javascript is case-sensitive.  Would be odd to not care in this case, but perhaps it doesn't.

Reply:
The `/i` at the end of the `/core operational – assets that are functionally/` marks it as case insensitive!